api/src/controllers/cohesion-center.js

```diff
@@ -230,19 +237,50 @@ router.get("/young/:youngId", passport.authenticate(["referent", "young"], { ses
   }
 });
 
-router.put("/", passport.authenticate("referent", { session: false, failWithError: true }), async (req, res) => {
+router.put("/:id", passport.authenticate("referent", { session: false, failWithError: true }), async (req, res) => {
   try {
-    const { error, value: newCenter } = validateUpdateCohesionCenter(req.body);
-    if (error) return res.status(400).send({ ok: false, code: ERRORS.INVALID_PARAMS, error: error.message });
+    const { error: errorId, value: checkedId } = validateId(req.params.id);
+    if (errorId) return res.status(400).send({ ok: false, code: ERRORS.INVALID_BODY, error });
 
     if (!canCreateOrUpdateCohesionCenter(req.user)) return res.status(403).send({ ok: false, code: ERRORS.OPERATION_UNAUTHORIZED });
 
-    const center = await CohesionCenterModel.findById(newCenter._id);
+    const { error, value: newCenter } = validateUpdateCohesionCenter(req.body);
+    if (error) return res.status(400).send({ ok: false, code: ERRORS.INVALID_PARAMS, error: error.message });
+
+    const center = await CohesionCenterModel.findById(checkedId);
     if (!center) return res.status(404).send({ ok: false, code: ERRORS.NOT_FOUND });
 
+    const previousCohorts = center.cohorts || [];
     center.set(newCenter);
     await center.save();
 
+    // if we change the cohorts, we need to update the sessionPhase1
+    if (newCenter?.cohorts?.length) {
+      const addedCohorts = newCenter.cohorts.filter((cohort) => !previousCohorts.includes(cohort));
+      const deletedCohorts = previousCohorts.filter((cohort) => !newCenter.cohorts.includes(cohort));
+
+      // add sessionPhase1 documents linked to this cohesion center
+      if (addedCohorts?.length > 0) {
+        for (let cohort of addedCohorts) {
+          const cohesionCenterId = center._id;
+          const placesTotal = center.placesTotal;
+          const placesLeft = center.placesTotal;
+          // :warning: peut etre creation de doublons ?
+          // il faudrait peut etre check si une session existe deja pour ce centre et cette cohorte avant de creer une nouvelle
+          await SessionPhase1.create({ cohesionCenterId, cohort, placesTotal, placesLeft });
```

Comment:
Oui ça me parait mieux ! Suggestion non testée :

```suggestion
          const session = await SessionPhase1.findOne({ cohesionCenterId, cohort });
          if (session) {
            session.set({ placesTotal, placesLeft })
            await session.save();
          } else {
            await SessionPhase1.create({ cohesionCenterId, cohort, placesTotal, placesLeft });
          }
```